### DISCUSSION TITLE
shipFromSupplySourceId - Where do we get the ID to fill the field?

### DISCUSSION THREAD

**Discussion by bullfrogsoftware** [amzn/selling-partner-api-models#4000] [General]
No documentation leads us to what piece of information we need to use to specify the warehouse we are shipping from.  Anyone know?

POST https://sellingpartnerapi-na.amazon.com/orders/v0/orders/123-1234567-1234567/shipmentConfirmation 
{  
   "marketplaceId": "ATVPDKIKX0DER",  
   "codCollectionMethod": "",  
   "packageDetail": {  
   "packageReferenceId": "123",  
   "carrierCode": "UPS",  
   "carrierName": "UPS",  
   "shippingMethod": "SHIPPING",  
   "trackingNumber": "1Z86V8030385598957",  
   "shipDate": "2022-11-30T16:15:30Z",  
   "shipFromSupplySourceId": "057d3fcc-b750-419f-bbcd-4d340c60c430",  <----- This ID ... ??
   "orderItems":[{  
     "orderItemId": "60696125413094",  
     "quantity": 1  
     }]  
   }  
 }

**Comment by tang226** [comment]
I encountered the same problem, please help!

**Comment by jawheral** [comment]
Hi @bullfrogsoftware,

The _shipFromSupplySourceId_ is optional. However, if you have configured SupplySources, you can use the Supply Sources API in order to configure and get the id from the supply source.

Please visit the API page here: https://developer-docs.amazon.com/sp-api/docs/supply-sources-api-v2020-07-01-reference

Best,
Jawher
Selling Partner API Solutions Architect

**Reply by tang226** [reply]
I have tried this but it doesn't work

**Reply by tang226** [reply]
Thank you for this. We completely overlooked that library. However, now we are confirming shipments on orders using the newly acquired GUID ID values, and the 'ship from' on the resultant package does not seem to change, at all. It is always set to the default warehouse. Any experience with this?

**Reply by jawheral** [reply]
@tang226 could you elaborate on what you refer to by "newly acquired GUID ID values"?

**Reply by tang226** [reply]
![3141723453494_ pic](https://github.com/user-attachments/assets/e3495566-53f0-48a9-9475-abc73c5d2b02)

supplySourceId

**Reply by tang226** [reply]
Actually I have already filed many cases, but they have not been resolved. If you need, I can forward the original email to you so that this issue can be resolved.
![3151723454096_ pic](https://github.com/user-attachments/assets/87417d03-cee3-4394-9c4d-33628f26a413)

**Reply by jawheral** [reply]
@tang226 thanks for sharing.
We can't act on these cases, please contact seller/developer support to follow up on those.

If you however, face any issues with the APIs, feel free to create an issue here on Github.

Best,
Jawher 
Selling Partner Solutions Architect

**Reply by tang226** [reply]
OK, the problem now is that the shipFromSupplySourceId parameter does not work when using the "/orders/v0/orders/{orderId}/shipmentConfirmation" interface, as described in my email.

"bullfrogsoftware" also encountered the same problem as me, and I saw his question in other forums
![3161723455107_ pic](https://github.com/user-attachments/assets/0348d67d-0419-4a5b-985b-2ca934625821)

**Reply by jawheral** [reply]
@tang226 Do I understand correctly that no matter which **_supplySourceId_** you provide, the shipment will always use a different supply source id which is the default warehouse/supplySource?

**Reply by tang226** [reply]
Yes, as you understand.

**Reply by jawheral** [reply]
Hi @tang226,

Our teams are aware and still working on this case. You will receive updates once there is progress.

I hope this helps!

Best,
Jawher
Selling Partner API Solutions Architect

**Reply by tang226** [reply]
> Hi @tang226,
> 
> Our teams are aware and still working on this case. You will receive updates once there is progress.
> 
> I hope this helps!
> 
> Best, Jawher Selling Partner API Solutions Architect

Hi, is there any result?

**Reply by jawheral** [reply]
Hi @tang226,

There's still no update yet. I will write an update here once available.

**Comment by bullfrogsoftware** [comment]
Yes, the parameter seems to be completely ignored.  It does not matter
which supply source GUID we apply to the call, the default is chosen in the
shipment.

On Mon, Aug 12, 2024 at 5:09 AM jawherallani ***@***.***>
wrote:

> Hi @tang226 <https://github.com/tang226>,
>
> Our teams are aware and still working on this case. You will receive
> updates once there is progress.
>
> I hope this helps!
>
> Best,
> Jawher
> Selling Partner API Solutions Architect
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/amzn/selling-partner-api-models/discussions/4000#discussioncomment-10312318>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGHL26O4KDVVPBAX22YCY4LZRCCUHAVCNFSM6AAAAABKH57SCCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMZRGIZTCOA>
> .
> You are receiving this because you were mentioned.Message ID:
> <amzn/selling-partner-api-models/repo-discussions/4000/comments/10312318@
> github.com>
>


-- 
Jeremy Clark
Bullfrog Software
***@***.***
432.235.0070 (office/voice)
432.235.0037 (text/sms)
http://www.bullfrogsoft.com
http://www.linkedin.com/in/bullfrogsoftware

This email and any files transmitted with it are confidential and intended
solely for the use of the individual or entity to whom they are addressed.
If you have received this email in error please notify the system manager.
This message contains confidential information and is intended only for the
individual named. If you are not the named addressee you should not
disseminate, distribute or copy this email. Please notify the sender
immediately by email if you have received this email by mistake and delete
this email from your system. If you are not the intended recipient you are
notified that disclosing, copying, distributing or taking any action in
reliance on the contents of this information is strictly prohibited.

**Comment by gaochenyue** [comment]
We are experiencing the same issue. Any update?

**Comment by vijaysolanki10** [comment]
The issue still persist. @jawheral 3 months since your teams being aware of this case and not update yet.

**Comment by bullfrogsoftware** [comment]
A case opened with Amazon revealed some information.  They indicated that
sending the  shipFromSupplySourceId value DOES actually write the value to
the database with the package, but the link between the data and the Seller
Central display screen is broken or not developed, so the correct value
will not be reflected when you go to check it.

On Tue, Dec 10, 2024 at 1:31 PM vijaysolanki10 ***@***.***>
wrote:

> The issue still persist. @jawheral <https://github.com/jawheral> 3 months
> since your teams being aware of this case and not update yet.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/amzn/selling-partner-api-models/discussions/4000#discussioncomment-11525054>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGHL26IFW3WOTZYSA5IA2B32E46QPAVCNFSM6AAAAABKH57SCCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNJSGUYDKNA>
> .
> You are receiving this because you were mentioned.Message ID:
> <amzn/selling-partner-api-models/repo-discussions/4000/comments/11525054@
> github.com>
>


-- 
Jeremy Clark
Bullfrog Software
***@***.***
432.235.0070 (office/voice)
432.235.0037 (text/sms)
http://www.bullfrogsoft.com
http://www.linkedin.com/in/bullfrogsoftware

This email and any files transmitted with it are confidential and intended
solely for the use of the individual or entity to whom they are addressed.
If you have received this email in error please notify the system manager.
This message contains confidential information and is intended only for the
individual named. If you are not the named addressee you should not
disseminate, distribute or copy this email. Please notify the sender
immediately by email if you have received this email by mistake and delete
this email from your system. If you are not the intended recipient you are
notified that disclosing, copying, distributing or taking any action in
reliance on the contents of this information is strictly prohibited.

**Comment by tang226** [comment]
You can try using the feed api, which solves my problem

**Comment by tang226** [comment]
You can try using the feed api, which solves my problem

https://developer-docs.amazon.com/sp-api/docs/order-feed-type-values
Order Fulfillment feed

**Reply by vijaysolanki10** [reply]
Can you please confirm what content-type did you use? json or xml?

I am more familiar with json so I used json but get 403 ERROR.
